### PR TITLE
Bind9 refused

### PR DESCRIPTION
**Pull request by mstilkerich**
This is a port of the fail2ban named-refused jail.

This is my first custom configuration for crowdsec, I appreciate feedback on things that can be improved!

### CODE REVIEW

File: collections/mstilkerich/bind9.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Acquisition configuration depends on whether bind9 is configured to log to
+syslog, separate log files, or both.
+
+For a separate log file, set the log type to `named`:
+
+```yaml
+filenames:
+  - /var/log/named/security.log
+labels:
+  type: named
```
Comment (LaurenceJJones) [inline collections/mstilkerich/bind9.md:22]: If type is set to syslog is the program name set the `named` within syslog?

File: parsers/s01-parse/mstilkerich/bind9-logs.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      pattern: '^%{BIND9_PREFIX}bad zone transfer request: ''\S+/IN'': non-authoritative zone (?:denied|\(NOTAUTH\))\s*$'
+      apply_on: message
+  - grok:
+      pattern: '^%{BIND9_PREFIX}(?:view (?:internal|external): )?query(?: \(cache\))? .* (?:denied|\(NOTAUTH\))\s*$'
+      apply_on: message
+statics:
+  - meta: service
+    value: bind9
+  - meta: log_type
+    value: bind9_denied
```
Comment (LaurenceJJones) [inline parsers/s01-parse/mstilkerich/bind9-logs.yaml:31]: This is okay at the moment, however, if we support multiple patterns they might need to be applied under each GROK pattern

**Review by LaurenceJJones — COMMENTED** [review]
Pending above questions

File: collections/mstilkerich/bind9.md (hunk 1 above)
Comment (mstilkerich) [reply collections/mstilkerich/bind9.md:22]: Yes, the bind daemon binary is called `named` and uses this also as the tag in syslog messages. Therefore, when log messages are taken from syslog the `syslog-logs` parser will set `Parsed.program` to `named`. Therefore I also use it as label for the bind9 logs, which the `non-syslog` parser will use for setting `Parsed.program` and subsequently the `bind9-logs` parser will see compatible events independent of whether they originate from syslog or a bind9 log file.

File: parsers/s01-parse/mstilkerich/bind9-logs.yaml (hunk 2 above)
Comment (mstilkerich) [reply parsers/s01-parse/mstilkerich/bind9-logs.yaml:31]: I don't understand what you mean by supporting multiple patterns (there already are multiple patterns in the parser), but I also take you don't currently see an issue here. Anyway I saw no reason to tag the different policy violation event types differently and therefore all are tagged as `bind9_denied` type.

**Comment by mstilkerich** [comment]
Thanks for your feedback, I have commented on your remarks.

**Pull request merged**